src/block/Block.php

```diff
@@ -427,14 +427,20 @@ public function getBreakInfo() : BlockBreakInfo{
 	 * @param Item[] &$returnedItems Items to be added to the target's inventory (or dropped, if full)
 	 */
 	public function onBreak(Item $item, ?Player $player = null, array &$returnedItems = []) : bool{
-		$world = $this->position->getWorld();
-		if(($t = $world->getTile($this->position)) !== null){
-			$t->onBlockDestroyed();
-		}
-		$world->setBlock($this->position, VanillaBlocks::AIR());
+		$this->onDestroy();
```

Comment:
Overriders of onBreak might not call this.

Reply:
Overriders always call and should call parent, however currently if they do not call you they will not replace the block with air.

So if they don't call parent, it will be their fault.

Reply:
They may replace it with air themselves, assuming that that's all the parent is doing. In such case, onDestroy() won't be called.

Reply:
That has been solved by moving the call of `Block::onDestroy()` to `World::destroyBlockInternal()`